source/base/utilities.cc

```diff
@@ -484,6 +484,37 @@ namespace Utilities
 
 
 
+  template <typename Number>
+  Number
+  truncate_to_n_digits(const Number number, const unsigned int n_digits)
+  {
+    AssertThrow(n_digits >= 1, ExcMessage("invalid parameter."));
+
+    if (!(std::fabs(number) > std::numeric_limits<Number>::min()))
+      return number;
+
+    const int order =
+      static_cast<int>(std::floor(std::log10(std::fabs(number))));
+
+    const int max_order = static_cast<int>(
+      std::floor(std::log10(std::numeric_limits<Number>::max())));
+
+    const int shift = -order + static_cast<int>(n_digits) - 1;
+
+    AssertThrow(
+      shift <= max_order,
+      ExcMessage(
+        "Overflow. Use a smaller value for n_digits and/or make sure "
+        "that the absolute value of number does not become too small."));
```

Comment:
Make clear that you're talking about a particular variable:
```suggestion
        "that the absolute value of 'number' does not become too small."));
```